### ISSUE TITLE
Not able to mask the payload and header with secret() method for REST and setCookie so that they are displayed in Allure report

### ISSUE THREAD

**Issue by AugustineAutoGit** [Codeception/CodeceptJS#4622]
#### What are you trying to achieve?
Need to mask the payload and Headers in REST and setCookie that may contain sensitive information

#### What do you get instead?
Not able to mask the payload and header with secret() method. Hence the value is shown in Allure report.
<img width="708" alt="image" src="https://github.com/user-attachments/assets/a2b0b58f-9405-45f0-834d-1f3dcb503803">

**Comment by AugustineAutoGit** [comment]
@kobenguyent could you please have a look?

**Comment by kobenguyent** [comment]
hey @AugustineAutoGit per my knowledge, that thing shall be resolved by allure plugin quickly. Not sure if this shall be handled by codeceptjs, as those test files are generated by allure plugin.

**Comment by AugustineAutoGit** [comment]
hi @kobenguyent 
But those steps are generated by CodeceptJS, like Allure plugin receives them from CodeceptJS only right?
if the secret() method masking was working fine, then Allure will not get unmasked values.
Also while in running with '--verbose', secret method is not masking as value are seen in logs as well.
In the below screenshot I masked the payload, like 'I.sendPostRequest('/api/users.json', secret({ "email": "user@user.com" }));', but it's still visible
<img width="1071" alt="image" src="https://github.com/user-attachments/assets/46d9fd4a-4b51-42e3-b04e-0ba35177b8b4">

**Comment by kobenguyent** [comment]
I could be wrong, but I think that, the proper way is that codeceptjs exposes the Step info as it is, the logs, plugins, etc shall mask the sensitive data as they desire. what do you think @DavertMik @AugustineAutoGit ?

**Comment by AugustineAutoGit** [comment]
> I could be wrong, but I think that, the proper way is that codeceptjs exposes the Step info as it is, the logs, plugins, etc shall mask the sensitive data as they desire. what do you think @DavertMik @AugustineAutoGit ?

@kobenguyent 
yes, but issue I am facing is the masking is not happening for REST

**Comment by AugustineAutoGit** [comment]
@kobenguyent meanwhile do you have any boilerplate example repository which have used CodeceptJS with Feature files(Gherkin) and Playwright along with Allure report: allure-codeceptjs not the legacy: @codeceptjs/allure-legacy

**Comment by gkushang** [comment]
+100 to this request. The `output` library prints the Tokens when `I.sendGetRequest` or similar function is called with `--verbose` or `--debug` flag. Even with `secret` function, it's printed

**Comment by AugustineAutoGit** [comment]
> +100 to this request. The `output` library prints the Tokens when `I.sendGetRequest` or similar function is called with `--verbose` or `--debug` flag. Even with `secret` function, it's printed

@gkushang 
do you have any boilerplate example repository which have used CodeceptJS with Feature files(Gherkin) and Playwright along with Allure report: allure-codeceptjs not the legacy: @codeceptjs/allure-legacy
?

**Comment by kobenguyent** [comment]
Closing as Allure team addressed this already. See https://github.com/allure-framework/allure-js/pull/1208
